cornflow/tests/unit/test_roles.py

```diff
@@ -311,22 +335,32 @@ def test_delete_user_role_authorized_user(self):
             self.assertEqual(200, response.status_code)
 
     def test_delete_and_create_user_role_authorized_user(self):
-        # TODO: this test should fail. But passes.
         role = self.roles_with_access[0]
         data = {
             "name": "testuser" + str(role),
             "email": "testemail" + str(role) + "@test.org",
             "password": "testpassword",
         }
         user_response = self.create_user(data)
-        user_role = self.create_role(user_response.json["id"], role)
+        planner_role = UserRoleModel.query.filter_by(
+            user_id=user_response.json["id"]
+        ).first()
+
+        self.create_role(user_response.json["id"], role)
+
         self.client.delete(
-            self.url + str(user_role.id) + "/",
+            self.url + str(planner_role.id) + "/",
```

Comment:
shouldn't it be `planner_role.role_id`?

`planner_role.id` returns the id of the row, not the id of the role. If I understand correctly.

Reply:
mmm i don't think it makes sense to have the endpoint require the id of the assignment. It should require "role+user". The internal assignment id should be unknown to the client. I think then you should change the endpoint to require user_id/role_id for the put / getOne / delete.

Reply:
Ok, will change it so it expects the user_id and the role_id